### PR TITLE
Core decrypt+encrypt methods

### LINKED ISSUE #341: Core decryptFile + encryptFile methods
part of #275, do before #195 or #196

 need to add the following method to Core. Roughly:

```swift
    public func decryptFile(encrypted: Data, keys: [PrvKeyInfo], msgPwd: String?) throws -> CoreRes.DecryptFile {
                let json: [String : Any?]? = ["keys": try keys.map { try $0.toDict() }, "msgPwd": msgPwd]
                let decrypted = try call("decryptFile", jsonDict: json, data: encrypted)
                let meta = try decrypted.json.decodeJson(as: CoreRes.DecryptFileMeta.self)
                return CoreRes.DecryptFile(...) 
     }
```

here is the code of the typescript handler:

```ts
  public decryptFile = async (uncheckedReq: any, data: Buffers): Promise<Buffers> => {
    const { keys: kisWithPp, msgPwd } = ValidateInput.decryptFile(uncheckedReq);
    const decryptedMeta = await PgpMsg.decrypt({ kisWithPp, encryptedData: Buf.concat(data), msgPwd });
    if (!decryptedMeta.success) {
      decryptedMeta.message = undefined;
      return fmtRes(decryptedMeta);
    }
    return fmtRes({ success: true, name: decryptedMeta.filename || '' }, decryptedMeta.content);
  }
```

need to be mindful of ability to handle the error scenario.

-----------------

similarly for encryptFile:

```
  public encryptFile = async (uncheckedReq: any, data: Buffers): Promise<Buffers> => {
    const req = ValidateInput.encryptFile(uncheckedReq);
    const encrypted = await PgpMsg.encrypt({ pubkeys: req.pubKeys, data: Buf.concat(data), filename: req.name, armor: false }) as OpenPGP.EncryptBinaryResult;
    return fmtRes({}, encrypted.message.packets.write());
  }
```

### PR DESCRIPTION
**Pull request by ekievsky**
This PR adds decrypt and encrypt file methods/tests to Core.

close #341

----------------------------------

**Tests** :
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCryptAppTests/Core/FlowCryptCoreTests.swift [hunk 1]
```diff
@@ -144,6 +144,118 @@ class FlowCryptCoreTests: XCTestCase {
...
+        do {
+            _ = try self.core.decryptFile(
+                encrypted: fileData,
+                keys: keys,
+                msgPwd: nil
+            )
+        } catch let CoreError.format(message) {
+            // Then
+            XCTAssertNotNil(message.range(of: "Error: Error during parsing"))
+        }
```
Comment (tomholub) [inline FlowCryptAppTests/Core/FlowCryptCoreTests.swift:223]: It seems this test will pass even if it didn't throw

File: FlowCryptAppTests/Core/FlowCryptCoreTests.swift [hunk 2]
```diff
@@ -144,6 +144,118 @@ class FlowCryptCoreTests: XCTestCase {
...
+                name: initialFileName
+            )
+            _ = try self.core.decryptFile(
+                encrypted: encrypted.encryptedFile,
+                keys: [],
+                msgPwd: nil
+            )
+        } catch let CoreError.keyMismatch(message) {
+            // Then
+            XCTAssertNotNil(message.range(of: "Missing appropriate key"))
```
Comment (tomholub) [inline FlowCryptAppTests/Core/FlowCryptCoreTests.swift:256]: Same here - badly designed test that will pass either way

**Review by tomholub — COMMENTED** [review]
See comments, else good

File: FlowCryptAppTests/Core/FlowCryptCoreTests.swift (hunk 1 above)
Comment (ekievsky) [reply FlowCryptAppTests/Core/FlowCryptCoreTests.swift]: good catch, thanks

**Review by tomholub — APPROVED** [review]

**Pull request merged**